ADR/0047-caching-for-container-build-layers.md

```diff
@@ -0,0 +1,33 @@
+# 47. Caching for container base images used during builds
+
+Date: 2025-06-22
+
+## Status
+
+Accepted
+
+## Context
+
+Konflux builds container images using tools like Buildah. During the build process, base images are fetched from container registries like `quay.io`. This repeated fetching of the same images results in high network traffic, slower build times, and exposes builds to failures caused by network issues or registry rate-limiting. This negatively impacts both the user experience (in terms of build speed and reliability) and operational costs of running the platform.
+
+## Decision
+
+We will implement a caching layer for container base images used during the build process using Squid HTTP proxies to mitigate these issues.
+
+* **Deployment:** The proxies will be deployed via a configurable Helm chart into a dedicated `proxy` namespace. A minimal, non-caching "passthrough" configuration will be part of the konflux-ci/konflux-ci "core" deployment to ensure systems are not prone to failure or security issues due to the proxy being unavalable.
+* **Access:** A well-known service endpoint, `http.proxy.svc.cluster.local`, will provide access to the proxy service within the cluster.
+* **TLS Caching:** We will use Squid's `ssl-bump` feature to cache content from TLS-encrypted connections. This requires establishing an internal Certificate Authority (CA) using `cert-manager`. Following the common practice of using separate CAs for separate concerns, the CA will be dedicated to the proxy service. This may be revisited in the future if a service mesh with a shared CA infrastructure (e.g. SPIFFE/SPIRE) is adopted.
+* **Trust:** The internal CA's trust bundle will be distributed to build pods using `trust-manager`.
+* **Integration:** The `buildah` Tekton task will be updated to use the proxy via the standard `http_proxy` and `https_proxy` environment variables. Users can opt-out by either configuring the `HTTP_PROXY` parameter to an empty string or by deny-listing paticular domains in the `no_proxy` environment variable via the `NO_PROXY` task parameter.
```

Comment:
In my opinion it should be explicit that the proxy is used. We can opt-in all users by default, but it still should be cleanly visible option and not hidden by internal defaults.